lib/backpex/live_resource.ex

```diff
@@ -248,806 +245,28 @@ defmodule Backpex.LiveResource do
-          options = Map.put(query_options, :filters, default_filter_options)
-          to = Router.get_path(socket, __MODULE__, params, :index, options)
-          push_navigate(socket, to: to)
-        else
-          socket
-        end
-      end
-
-      defp maybe_redirect_to_default_filters(socket) do
-        socket
-      end
-
-      defp maybe_put_search(query_options, %{"search" => search} = _params)
-           when is_nil(search) or search == "",
-           do: query_options
-
-      defp maybe_put_search(query_options, %{"search" => search} = _params),
-        do: Map.put(query_options, :search, search)
-
-      defp maybe_put_search(query_options, params), do: query_options
-
-      def assign_items(socket) do
-        %{
-          live_resource: live_resource,
-          fields: fields
-        } = socket.assigns
-
-        criteria = build_criteria(socket.assigns)
-        items = Resource.list(fields, socket.assigns, live_resource, criteria)
-
-        assign(socket, :items, items)
-      end
-
-      defp maybe_assign_metrics(socket) do
-        %{
-          assigns:
-            %{
-              repo: repo,
-              schema: schema,
-              live_action: live_action,
-              live_resource: live_resource,
-              params: params,
-              fields: fields,
-              query_options: query_options,
-              metric_visibility: metric_visibility
-            } = assigns
-        } = socket
-
-        filters = Backpex.LiveResource.get_active_filters(__MODULE__, assigns)
-
-        metrics =
-          metrics()
-          |> Enum.map(fn {key, metric} ->
-            query =
-              EctoAdapter.list_query(
-                assigns,
-                &item_query(&1, live_action, assigns),
-                fields,
-                search: search_options(query_options, fields, schema),
-                filters: filter_options(query_options, filters)
-              )
-
-            case Backpex.Metric.metrics_visible?(metric_visibility, live_resource) do
-              true ->
-                data =
-                  query
-                  |> Ecto.Query.exclude(:select)
-                  |> Ecto.Query.exclude(:preload)
-                  |> Ecto.Query.exclude(:group_by)
-                  |> metric.module.query(metric.select, repo)
-
-                {key, Map.put(metric, :data, data)}
-
-              _visible ->
-                {key, metric}
-            end
-          end)
-
-        socket
-        |> assign(metrics: metrics)
-      end
+      def handle_params(params, url, socket), do: LiveResource.handle_params(params, url, socket)
 
       @impl Phoenix.LiveView
-      def handle_event("close-modal", _params, socket) do
-        socket =
-          socket
-          |> push_patch(to: socket.assigns.return_to)
-
-        {:noreply, socket}
-      end
+      def handle_event(event, params, socket), do: LiveResource.handle_event(event, params, socket)
```

Comment:
@pehbehbeh this will catch all events, and users are no longer able to define own `handle_event` functions in their LiveResources. Same with `handle_info` etc.